### PR TITLE
Raise the default power levels for invites, tombstones and server acls

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR makes the following changes to the default power levels:

* Raises invites to be PL 50
* Raises tombstones to be PL 100
* Raises server ACLs to be PL 100

Sytests: https://github.com/matrix-org/sytest/pull/805

### CODE REVIEW

File: synapse/handlers/room.py [hunk 1]
```diff
@@ -825,19 +828,26 @@ def send(etype, content, **kwargs):
...
-                "invite": 0,
+                "invite": 50,
             }
 
             if config["original_invitees_have_ops"]:
                 for invitee in invite_list:
                     power_level_content["users"][invitee] = 100
 
+            # Allow power levels to be defined per chat preset
+            preset_power_level_override = config.get("power_level_content_override")
```
Comment (richvdh) [inline synapse/handlers/room.py]: we should probably decide that either `power_level_content_override` is an optional property for the preset dict, and remove the no-op settings, or decide that it is compulsory, and make this flow unconditional.

I'd vote for the latter, but I don't really mind.

File: synapse/handlers/room.py [hunk 2]
```diff
@@ -825,19 +828,26 @@ def send(etype, content, **kwargs):
...
+            if preset_power_level_override:
+                power_level_content.update(preset_power_level_override)
```
Comment (richvdh) [inline synapse/handlers/room.py]: bit of a danger of doing this like this is that when we want to override the PL for a single event type, the obvious approach won't work. I think it's ok though.

File: synapse/handlers/room.py [hunk 3]
```diff
@@ -64,18 +64,21 @@ class RoomCreationHandler(BaseHandler):
...
             "original_invitees_have_ops": True,
             "guest_can_join": True,
+            "power_level_content_override": {},
         },
         RoomCreationPreset.PUBLIC_CHAT: {
             "join_rules": JoinRules.PUBLIC,
             "history_visibility": "shared",
             "original_invitees_have_ops": False,
             "guest_can_join": False,
+            "power_level_content_override": {"invite": 50},
```
Comment (richvdh) [inline synapse/handlers/room.py]: isn't this the new default? I'm confused.

**Comment by aaronraimist** [comment]
Why is invite changing?

**Comment by richvdh** [comment]
@aaronraimist: spam, in a word.

**Comment by jplatte** [comment]
> @aaronraimist: spam, in a word.

Is there a detailed explanation about this somewhere?

**Comment by richvdh** [comment]
> Is there a detailed explanation about this somewhere?

From an internal issue:

> This means that an invite spammer can sabotage a public room by joining it and then inviting thousands of users to it, encouraging moderators to shutdown the room to mitigate the spam.

**Pull request merged**